snuba/utils/streams/consumer.py

```diff
@@ -290,6 +303,63 @@ def unsubscribe(self) -> None:
 
         self.__consumer.unsubscribe()
 
+    # Managed Consumer Methods
+
+    def assign(
+        self, partitions: Mapping[Partition, Optional[int]]
+    ) -> Mapping[Partition, int]:
+        """
+        Set the current partition assignment. This replaces any existing
+        assignment.
+
+        If no offset is provided, the last committed offset is used. If there
+        is no committed offset stored, the strategy specified by the
+        ``auto.offset.reset`` configuration parameter is used to determine
+        the starting offset.
+
+        Raises an ``InvalidState`` exception if called on a closed consumer.
+        """
+        if self.__state is not KafkaConsumerState.CONSUMING:
+            raise InvalidState(self.__state)
+
+        assignment = {
+            **{
+                partition: offset
+                for partition, offset in partitions.items()
+                if offset is not None
+            },
+            **self.__get_starting_offsets(
+                [
+                    partition
+                    for partition, offset in partitions.items()
+                    if offset is None
+                ]
+            ),
+        }
+
+        self.__consumer.assign(
+            [
+                ConfluentTopicPartition(partition.topic.name, partition.index, offset)
+                for partition, offset in assignment.items()
+            ]
+        )
```

Comment:
What happens if I try to assign an offset lower than `earliest` or higher than `latest` ? Is it suppsoed to throw or to let Confluent handle this case ?